pkg/services/govmomi/vcenter/clone.go

```diff
@@ -229,27 +218,31 @@ func Clone(ctx *context.VMContext, bootstrapData []byte, format bootstrapv1.Form
 			return errors.Wrapf(err, "unable to get storageProfileID from name %s for %q", ctx.VSphereVM.Spec.StoragePolicyName, ctx)
 		}
 
-		kind := []string{"Datastore"}
-		m := view.NewManager(ctx.Session.Client.Client)
-
-		v, err := m.CreateContainerView(ctx, ctx.Session.Client.Client.ServiceContent.RootFolder, kind, true)
-		if err != nil {
-			return errors.Wrapf(err, "unable to create container view for Datastore for %q", ctx)
-		}
-
-		var content []types.ObjectContent
-		err = v.Retrieve(ctx, kind, []string{"name"}, &content)
-		_ = v.Destroy(ctx)
-		if err != nil {
-			return errors.Wrapf(err, "unable to retrieve container view for Datastore for %q", ctx)
-		}
-
 		var hubs []pbmTypes.PbmPlacementHub
-		for _, ds := range content {
+
+		// If there's a datastore request, it should be the only one requested from Storage Policy
+		if datastoreRef != nil {
 			hubs = append(hubs, pbmTypes.PbmPlacementHub{
-				HubType: ds.Obj.Type,
-				HubId:   ds.Obj.Value,
+				HubType: datastoreRef.Type,
+				HubId:   datastoreRef.Value,
 			})
+		} else {
+			// Otherwise we should get just the Datastores connected to our pool
+			cluster, err := pool.Owner(ctx)
```

Comment:
All in all the change looks good to me. Just one question for verification:

We always have a `pool` (which is a resourcePool right?) to use or are there cases where there could be no pool and the old variant would need to get applied?

(note: we use `ctx.Session.Finder.ResourcePoolOrDefault` so I guess we would always get a valid resource pool?)

Reply:
IIRC the pool will always exists. If there is no pool, it will return a default one as you stated. It should get from https://github.com/vmware/govmomi/blob/ba843e53ef62703cf15f0f263c003d720b63cc1e/find/finder.go#L924

I remember there was an issue we triaged this week on Resource Pools not existing on non DRS clusters, but I think this is already a current problem and in that case we should probably target clusters and not resource pools (like a long term problem)